src/hooks/useAuth.tsx

```diff
@@ -118,30 +123,39 @@ export const AuthContextProvider = ({
     [refreshAuthResult],
   );
 
-  const refreshIfNeeded = useCallback(async () => {
-    if (
-      !loading &&
-      refreshHandler &&
-      authResult &&
-      shouldAttemptTokenRefresh(authResult.accessTokenExpirationDate)
-    ) {
-      refreshAuthResult(refreshHandler, authResult);
-    }
-  }, [loading, refreshHandler, authResult, refreshAuthResult]);
+  const refreshIfNeeded = useCallback(
+    async (skipExpirationCheck: boolean = false) => {
+      if (
+        !loading &&
+        refreshHandler &&
+        authResult &&
+        (skipExpirationCheck ||
+          shouldAttemptTokenRefresh(authResult.accessTokenExpirationDate))
+      ) {
+        return refreshAuthResult(refreshHandler, authResult);
+      }
+    },
+    [loading, refreshHandler, authResult, refreshAuthResult],
+  );
 
   useEffect(() => {
     if (currentAppState === 'active') {
       refreshIfNeeded();
     }
   }, [currentAppState, refreshIfNeeded]);
 
+  const refreshForAuthFailure = useCallback(async () => {
```

Comment:
I feel like we might want a different name here (something like `refreshAuth` or `refresh`), since the current name does not convey that it will always refresh the token because it sets `skipExpirationCheck` to true and because it suggests how/when it should be used.

Reply:
Yea I wrestled with the naming of this method.  What I want this to be is is combo of "report auth failure" and "refresh token if not already refreshing - and if refreshing fails I expect isLoggedIn to change".  I started with the former, `reportAuthFailure` because I don't want to indicate consumers have more power than they do.  Note that `skipExpirationCheck=true` does NOT mean it will always refresh; namely if `loading=true` the invocation will be essentially ignored.  Lmk if you or others still feel a more straight forward method name is better here - if we go that route, I think we should just expose `refreshIfNeeded`.  I personally still like this extra method that abstracts the internals a bit.  I could see this method evolving on its own, e.g. with debouncing etc.